MyHashMap.java

```diff
@@ -0,0 +1,196 @@
+package com.company;
+
+import java.util.Objects;
+
+public class MyHashMap<K, V> implements MyMap<K, V> {
+    private Node[] nodesArr;
+    private double loadFactor = 0.75;
+    private int threshold = 12;
+    private int size = 16;
+    private int capacity;
+
+    MyHashMap() {
+        nodesArr = new MyHashMap.Node[size];
+    }
+
+    public void setLoadFactor(double loadFactor) {
+        this.loadFactor = loadFactor;
+    }
+
+    @Override
+    public V put(K key, V value) {
+        if (key != null) {
+            remove(key);
+            capacity++;
+            if (capacity >= threshold) {
+                resize();
+            }
+            putVal(hashIndex(key, nodesArr.length), value, key);
+            return value;
+        }
+        return null;
+    }
+
+    public int hashIndex(K key, int n) {
+        if (key == null) {
+            return 0;
+        }
+        return hashCode(key) & (n - 1);
+    }
+
+    public int hashCode(K key) {
+        return Objects.hashCode(key);
+    }
+
+    private void putVal(int index, V value, K key) {
+        if (isEmpty(nodesArr[index])) {
+            nodesArr[index] = new Node(hashCode(key), key, value, null);
+        } else {
+            Node currentNode = nodesArr[index];
+            while (!isEmpty(currentNode.next)) {
+                currentNode = currentNode.next;
+            }
+            currentNode.next = new Node(hashCode(key), key, value, null);
+        }
+    }
+
+    private void resize() {
+        int size = nodesArr.length;
+        Node[] tempNode = new MyHashMap.Node[size * 2];
+        System.arraycopy(nodesArr, 0, tempNode, 0, size);
+        this.nodesArr = tempNode;
+        this.size = size;
+        this.threshold = (int) (nodesArr.length * loadFactor);

```

Comment:
this not needed

Reply:
> this not needed

I do this because I think it makes the code clearer